sdk/src/main/scala/io/horizen/account/websocket/WebSocketAccountServerEndpoint.scala

```diff
@@ -0,0 +1,270 @@
+
+      case LOGS_SUBSCRIPTION.method =>
+        if (rpcParams.size() < 2)
+          WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+            new RpcError(RpcCode.InvalidParams, "Missing filters (address, topics).", "")),
+            session)
+        val subscriptionId = createSubscriptionId()
+        val filterQuery = EthJsonMapper.deserialize(rpcParams.get(1).toString, classOf[FilterQuery])
+        filterQuery.sanitize()
+        WebSocketAccountServerEndpoint.addLogsSubscription(SubscriptionWithFilter(session, subscriptionId,
+          filterQuery
+        ))
+        log.debug("New Subscription on logs "+session.getId)
+        WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, subscriptionId), session)
+
+      case unkownMethod =>
+        WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+          new RpcError(RpcCode.InvalidParams, "unsupported subscription type " + unkownMethod, "")),
+          session)
+        log.debug("unsupported subscription type " + unkownMethod)
+    }
+
+  }
+
+  private def unsubscribe(session: Session, rpcRequest: RpcRequest): Unit = {
+    val rpcParams = rpcRequest.params
+
+    val subscriptionIdsToRemove: util.ArrayList[BigInteger] = new util.ArrayList[BigInteger]()
+    rpcParams.forEach(subscriptionId => {
+      subscriptionIdsToRemove.add(Numeric.decodeQuantity(subscriptionId.asText()))
+    })
+    val removedSubscription = WebSocketAccountServerEndpoint.removeSubscriptions(subscriptionIdsToRemove.toArray(new Array[BigInteger](0)))
+    if (!removedSubscription) {
+      WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+        new RpcError(RpcCode.InvalidParams, s"Subscription ID not found.", "")),
+        session)
+    } else
+      WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, true), session)
+  }
+
+  private def validateRpcRequestParams(rpcParams: JsonNode, rpcRequest: RpcId, session: Session): Unit = {
+    if (rpcParams == null || !rpcParams.isArray || rpcParams.size() < 1) {
+      log.debug("Missing or empty field params.")
+      WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest,
+        new RpcError(RpcCode.InvalidParams, "Missing or empty field params.", "")),
+        session)
+    }
+  }
+
+  private def createSubscriptionId(): BigInteger = {
+    BigInteger.valueOf(WebSocketAccountServerEndpoint.subscriptionCounter.incrementAndGet())
+  }
+}
+
+private object WebSocketAccountServerEndpoint extends SparkzLogging {
+  var subscriptionCounter: AtomicInteger = new AtomicInteger(0)
+  var newHeadsSubscriptions: util.ArrayList[Subscription] = new util.ArrayList[Subscription]()
+  var newPendingTransactionsSubscriptions: util.ArrayList[Subscription] = new util.ArrayList[Subscription]()
+  var logsSubscriptions: util.ArrayList[SubscriptionWithFilter] = new util.ArrayList[SubscriptionWithFilter]()
```

Comment:
1) If possible let's use Scala data structures like ArrayBuffer
2) moreover all fields above can be `val`

Reply:
Done